### PR TITLE
Refactor: add AggregateDeclaration.newScope to factor out common code

### PR DESCRIPTION
**Pull request by 9rnsr**
The underlying concept is same with `AttribDeclaration.newScope`.

### CODE REVIEW

File: src/aggregate.h [hunk 1]
```diff
@@ -119,7 +119,7 @@ class AggregateDeclaration : public ScopeDsymbol
     Expression *getRTInfo;      // pointer to GC info generated by object.RTInfo(this)
 
     AggregateDeclaration(Loc loc, Identifier *id);
-    void setScope(Scope *sc);
```
Comment (yebblies) [inline src/aggregate.h]: ?

Comment (9rnsr) [inline src/aggregate.h]: It has removed in #5500 (https://github.com/dlang/dmd/commit/75b5b691555bef2d2f21b44821f297f93ddfbc77#diff-d9f55fb03991f402fab0f1fdec9effb2L116), but I had forgotten updating C++ header.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
Thanks.

**Pull request merged**